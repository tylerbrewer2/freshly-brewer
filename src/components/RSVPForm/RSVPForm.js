import React from 'react';

// Components
import Form from '../Form';
import TextInputGroup from '../TextInputGroup';
import TextAreaGroup from '../TextAreaGroup';

// Utilities
import AttendeeCreator from '../../utilities/AttendeeCreator';


export default class RSVPForm extends React.Component {
  state = {
    email: '',
    firstName: '',
    lastName: '',
    attending: '',
    extraAttendees: '',
    songSuggestion: '',
    message: '',
    response: undefined,
  }

  onChange = (e) => {
    const state = Object.assign({}, this.state)
    state[e.currentTarget.name] = e.currentTarget.value
    this.setState(prev => ({
      ...prev,
      ...state,
    }));
  }

  onSubmit = (e) => {
    e.preventDefault();
    const state = Object.assign({}, this.state)
    delete state.response
    const attendeeCreator = new AttendeeCreator(state);
    const response = attendeeCreator.create({ successMessage: 'RSVP successfully sent.' });
    this.setState(prev => ({
      ...prev,
      response,
    }));
  }

  render() {
    return (
      <section>
        <Form
          onSubmit={this.onSubmit}
          title="RSVP for December 2nd, 2017"
          width={6}
          requestResponse={this.state.response}
        >
          <TextInputGroup
            labelText="Email*"
            identifier="email"
            value={this.state.email}
            onChange={this.onChange}
            required
          />

          <TextInputGroup
            labelText="First Name*"
            identifier="firstName"
            value={this.state.firstName}
            onChange={this.onChange}
            required
          />

          <TextInputGroup
            labelText="Last Name*"
            identifier="lastName"
            value={this.state.lastName}
            onChange={this.onChange}
            required
          />

          <TextInputGroup
            labelText="Will you be attending?*"
            identifier="attending"
            value={this.state.attending}
            onChange={this.onChange}
            required
          />

          <TextInputGroup
            labelText="Who else is coming with you?"
            identifier="extraAttendees"
            value={this.state.extraAttendees}
            onChange={this.onChange}
          />

          <TextInputGroup
            labelText="Do you have any song suggestions?"
            identifier="songSuggestion"
            value={this.state.songSuggestion}
            onChange={this.onChange}
          />

          <TextAreaGroup
            labelText="Are you as excited as we are? Let us know!"
            identifier="message"
            value={this.state.message}
            onChange={this.onChange}
          />
        </Form>
      </section>
    )
  }
};

        // <TextAreaGroup
        //   labelText="Are you as excited as we are? Let us know!"
        //   identifier="message"
        //   value={this.state.message}
        //   onChange={this.onChange}
        // />