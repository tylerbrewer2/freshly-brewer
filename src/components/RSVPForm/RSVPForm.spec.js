import React from 'react';
import renderer from 'react-test-renderer';

import RSVPForm from './RSVPForm';

describe('component structure', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <RSVPForm />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
