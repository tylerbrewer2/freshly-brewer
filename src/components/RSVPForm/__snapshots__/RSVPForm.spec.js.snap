exports[`component structure renders correctly 1`] = `
<section>
  <div
    className="undefined  center-xs row">
    <div
      className="col-xs-6">
      <section
        className="sc-bxivhb eianJZ">
        <h1
          className="sc-bdVaJa dcMGOa">
          RSVP for December 2nd, 2017
        </h1>
        <form
          onSubmit={[Function]}>
          <div
            className="col-xs-12">
            <div
              className="sc-bZQynM bSnXeh">
              <label
                className="sc-ifAKCX iPUEyZ"
                for={undefined}>
                Email*
              </label>
              <input
                className="sc-EHOje iUexnX"
                id="email"
                name="email"
                onChange={[Function]}
                type="text"
                value="" />
            </div>
          </div>
          <div
            className="col-xs-12">
            <div
              className="sc-bZQynM bSnXeh">
              <label
                className="sc-ifAKCX iPUEyZ"
                for={undefined}>
                First Name*
              </label>
              <input
                className="sc-EHOje iUexnX"
                id="firstName"
                name="firstName"
                onChange={[Function]}
                type="text"
                value="" />
            </div>
          </div>
          <div
            className="col-xs-12">
            <div
              className="sc-bZQynM bSnXeh">
              <label
                className="sc-ifAKCX iPUEyZ"
                for={undefined}>
                Last Name*
              </label>
              <input
                className="sc-EHOje iUexnX"
                id="lastName"
                name="lastName"
                onChange={[Function]}
                type="text"
                value="" />
            </div>
          </div>
          <div
            className="col-xs-12">
            <div
              className="sc-bZQynM bSnXeh">
              <label
                className="sc-ifAKCX iPUEyZ"
                for={undefined}>
                Will you be attending?*
              </label>
              <input
                className="sc-EHOje iUexnX"
                id="attending"
                name="attending"
                onChange={[Function]}
                type="text"
                value="" />
            </div>
          </div>
          <div
            className="col-xs-12">
            <div
              className="sc-bZQynM bSnXeh">
              <label
                className="sc-ifAKCX iPUEyZ"
                for={undefined}>
                Who else is coming with you?
              </label>
              <input
                className="sc-EHOje iUexnX"
                id="extraAttendees"
                name="extraAttendees"
                onChange={[Function]}
                type="text"
                value="" />
            </div>
          </div>
          <div
            className="col-xs-12">
            <div
              className="sc-bZQynM bSnXeh">
              <label
                className="sc-ifAKCX iPUEyZ"
                for={undefined}>
                Do you have any song suggestions?
              </label>
              <input
                className="sc-EHOje iUexnX"
                id="songSuggestion"
                name="songSuggestion"
                onChange={[Function]}
                type="text"
                value="" />
            </div>
          </div>
          <div
            className="col-xs-12">
            <div
              className="sc-htoDjs bswhhT">
              <label
                className="sc-ifAKCX iPUEyZ"
                for={undefined}>
                Are you as excited as we are? Let us know!
              </label>
              <textarea
                className="sc-gzVnrw bYhlWP"
                id="message"
                name="message"
                onChange={[Function]}
                value="" />
            </div>
          </div>
          <div
            className="col-xs-12">
            <button
              className="sc-bwzfXH bOFSPc"
              type="submit">
              SUBMIT
            </button>
          </div>
        </form>
      </section>
    </div>
  </div>
</section>
`;
